src/code/z_actor.c

```diff
@@ -457,439 +1232,3482 @@ f32 Actor_XZDistanceToPoint(Actor* actor, Vec3f* point) {
+    f32 temp_f12;
+    s16 temp_v0 = BINANG_SUB(BINANG_SUB(actor->yawTowardsPlayer, 0x8000), angle);
+    s16 yaw = ABS_ALT(temp_v0);
+
+    if (player->unk_730 != NULL) {
+        if ((yaw > 0x4000) || ((actor->flags & ACTOR_FLAG_8000000))) {
+            return FLT_MAX;
+        }
+
+        temp_f12 = actor->xyzDistToPlayerSq - ((actor->xyzDistToPlayerSq * 0.8f) * ((0x4000 - yaw) * (1.0f / 0x8000)));
+        return temp_f12;
+    }
+
+    if (yaw >= 0x2AAB) {
+        return FLT_MAX;
+    }
+    return actor->xyzDistToPlayerSq;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800B8C50.s")
+TargetRangeParams D_801AECF0[] = {
+    TARGET_RANGE(70, 140),   TARGET_RANGE(170, 255),    TARGET_RANGE(280, 5600),      TARGET_RANGE(350, 525),
+    TARGET_RANGE(700, 1050), TARGET_RANGE(1000, 1500),  TARGET_RANGE(100, 105.36842), TARGET_RANGE(140, 163.33333),
+    TARGET_RANGE(240, 576),  TARGET_RANGE(280, 280000), TARGET_RANGE(2500, 3750),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800B8C78.s")
+s32 func_800B83BC(Actor* actor, f32 arg1) {
+    return arg1 < D_801AECF0[actor->targetMode].rangeSq;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800B8C9C.s")
+s32 func_800B83F8(Actor* actor, Player* player, s32 flag) {
+    if ((actor->update == NULL) || !(actor->flags & ACTOR_FLAG_1) || (actor->flags & ACTOR_FLAG_8000000)) {
+        return true;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800B8CEC.s")
+    if (!flag) {
+        s16 yaw = BINANG_SUB(actor->yawTowardsPlayer, 0x8000) - player->actor.shape.rot.y;
+        s16 phi_v1 = ABS_ALT(yaw);
+        f32 dist;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800B8D10.s")
+        if ((player->unk_730 == NULL) && (phi_v1 >= 0x2AAB)) {
+            dist = FLT_MAX;
+        } else {
+            dist = actor->xyzDistToPlayerSq;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800B8D50.s")
+        return !func_800B83BC(actor, D_801AECF0[actor->targetMode].leashScale * dist);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800B8D98.s")
+    return false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800B8DD4.s")
+s16 D_801AED48[] = {
+    0x101, 0x141, 0x111, 0x151, 0x105, 0x145, 0x115, 0x155,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800B8E1C.s")
+s32 Actor_ProcessTalkRequest(Actor* actor, GameState* gameState) {
+    if (actor->flags & ACTOR_FLAG_100) {
+        actor->flags &= ~ACTOR_FLAG_100;
+        return true;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800B8E58.s")
+    return false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/Audio_PlayActorSound2.s")
+// Actor_PickUpExchange? Seems to be called with exchangeItemId -1 if the same actor used Actor_PickUp
+s32 func_800B8500(Actor* actor, GameState* gameState, f32 xzRange, f32 yRange, s32 exchangeItemId) {
```

Comment:
Could you explain why this would take `GameState` and not `GlobalContext`? In particular `GET_PLAYER` would have UB if it is not `GlobalContext`, so to me it only makes sense to pass `GlobalContext`.

Reply:
I did this because of the hypothesis of "public api functions take GameState" instead of GlobalContext"

I can change if back if you think it would be better that way

Reply:
Personally, I do think it would be better the other way since any other `GameState` besides `GlobalContext` (`Play`) would have UB when getting player through the `actorCtx`. Though this is probably going to have to be discussed in Discord along with OOT (as I know they are also having the same discussions).  If it does stay as `GameState`, I'd prefer removing the globalCtx cast in the `Player` macro, and have an explicit inline cast here. The cast is too important to hide behind a macro imo.